Cerchie's post on text resizing in ios and android (#1188)
a11yproject/a11yproject.com: src/posts/2020-02-05-text-resizing-in-ios-and-android.md

---
layout: post
title: Text resizing in iOS and Android
description: Things to keep in mind for designing accessible mobile apps.
category: Mobile
author: Lucia Cerchie
date: 2021-01-28
further_reading:
  - title: BBC Accessibility Guide
    url: https://www.bbc.co.uk/accessibility/forproducts/guides/mobile/content-resizing/
    source: BBC
  - title: Designing Apps with Accessible Fonts
    url: https://www.mobileux.net/2019/06/24/designing-apps-with-accessible-fonts/
    source: Mobile UX
   
eleventyNavigation:
  key: {{ category }}
  title: {{ title }}
tags:
  - mobile
---

> When you're designing for mobile, it's important to remember that some people will need to change the size of the content
> to read it easily. There are so many ways to view content on mobile devices, from zooming in to orienting the screen differently, so a failure to design in a dynamic environment can lead to accessibility issues. There are different ways you can attend to this, whether you're writing an app for iOS or Android.

Reviewer: ```suggestion
When you're designing for mobile, it's important to remember that some people will need to change the size of the content to read it easily. 

There are so many ways to view content on mobile devices, from zooming in to orienting the screen differently, so a failure to design in a dynamic environment can lead to accessibility issues. There are different ways you can attend to this, whether you're writing an app for iOS or Android.
```

Author: Thanks!

> When you're designing for mobile, it's important to remember that some people will need to change the size of the content
> to read it easily. There are so many ways to view content on mobile devices, from zooming in to orienting the screen differently, so a failure to design in a dynamic environment can lead to accessibility issues. There are different ways you can attend to this, whether you're writing an app for iOS or Android.

Reviewer: ```suggestion
When you're designing for mobile, it's important to remember that some people will need to change the size of the content
to read it easily. There are so many ways to view content on mobile devices, from zooming in to orienting the screen differently, so a failure to design in a dynamic environment can lead to accessibility issues. There are different ways you can attend to this, whether you're making an app for iOS or Android.
```

Author: Thanks!

## Considerations for iOS

> In iOS, users can only zoom in on specified areas. To allow them to resize the type, use [dynamic type](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/#dynamic-type-sizes), which allows them to [resize the type](https://docs.google.com/spreadsheets/d/1B7UjuckBckAQtILeBCqoIUe3Ze3R8ow2KOQSM_JJI_w/edit#gid=0) in your app.

Reviewer: >In iOS, users can only zoom in on specified areas.

Is this true? You can zoom the entire viewport via the Zoom functionality, then pan around.

Author: I added a source for the difference between Zoom functionalities.

> In iOS, users can only zoom in on specified areas. To allow them to resize the type, use [dynamic type](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/#dynamic-type-sizes), which allows them to [resize the type](https://docs.google.com/spreadsheets/d/1B7UjuckBckAQtILeBCqoIUe3Ze3R8ow2KOQSM_JJI_w/edit#gid=0) in your app.

Reviewer: >[resize the type](https://docs.google.com/spreadsheets/d/1B7UjuckBckAQtILeBCqoIUe3Ze3R8ow2KOQSM_JJI_w/edit#gid=0)

Where is this spreadsheet coming from?

Author: Couldn't find where I found the resource, so I removed it

Reviewer: Thanks for that. Since it's a Google sheet we can't as reliably track when it gets updated or how valid it is.

> If you manually limit how large your text scales it may cause issues for people who need a bigger font. 

Reviewer: What kind of issues?

Author: Expounded here.

> [Minimum font-size](https://uxdesign.cc/guide-for-designing-better-mobile-apps-typography-5796495ef86f) should be 17pt.

Reviewer: Minimum font size for what?

Author: Added context.

> ## Considerations for Android

Reviewer: ```suggestion
## Considerations for Android

```

Author: Thanks!

Again, users can only zoom in on specified areas, so you'll need to use [scale-independent pixels](https://www.pixel-ruler.net/android-scale-independent-pixel) to determine the size of your text.

> There isn't a tool like Dynamic Type in Android, so you'll need to scale the text at different sizes yourself. Material design [has a resource](https://material.io/design/typography/the-type-system.html#type-scale) that makes a good starting point.

Reviewer: ```suggestion
There isn't a tool like Dynamic Type in Android, so you'll need to scale the text at different sizes yourself. [Material design has a resource](https://material.io/design/typography/the-type-system.html#type-scale) that makes a good starting point.
```

Adding some context to the link's purpose for users who may be navigating via a list of links.

Author: Thanks!

> [Minimum font-size](https://uxdesign.cc/guide-for-designing-better-mobile-apps-typography-5796495ef86f) should be 16sp:

Reviewer: Minimum font size for what?

Author: added context in my commit

> `android: textSize=16sp`

Reviewer: ```suggestion
```
android: textSize=16sp
```
```

This will use a full-width code fence area.

Author: I wasn't able to find the markdown in the git docs for this. I tried 
```android: textSize=16sp
``` but I'm getting a blank fence with that

Reviewer: Yeah, it's weird. I think this is an issue with nested fences, but the suggested code _should_ render properly when rendered.

You'll also want to remember to make your [content container scrollable](https://medium.com/mesmerhq/designing-accessible-text-for-android-variable-font-and-screen-sizes-392fd386aea5). If it's not, you'll just cut off the extra content when a person zooms in.

## HTML

> 

Reviewer: Can we have a sentence or two telling the reader that apps may contain HTML views?

Author: Check!

> In your HTML, make sure to use relative units for responsive design. 

Reviewer: Can we link to a list of relative units for those who may be unfamiliar?

Author: Link added!

> In your HTML, make sure to use relative units for responsive design. 

Reviewer: Why should we use relative units? What happens if we don't?

Author: added context in my new commits

> Note that a person's pinch-to-zoom capability will be frustrated on mobile < iOS10 if you add `maximum-scale=1.0` to the `meta` element.

Reviewer: Why will they be frustrated? Maybe we can link to: https://www.a11yproject.com/posts/2013-01-14-never-use-maximum-scale/

Author: Added this! And made the suggested edits for all the rest of your helpful comments, besides the codefence one.

  - mobile
---

When you're designing for mobile, it's important to remember that some people will need to change the size of the content to read it easily. 

There are so many ways to view content on mobile devices, from zooming in to orienting the screen differently, so a failure to design in a dynamic environment can lead to accessibility issues. There are different ways you can attend to this, whether you're writing an app for iOS or Android.

## Considerations for iOS

> In iOS, users can use [Full Screen Zoom or Window Zoom](https://support.apple.com/guide/iphone/zoom-iph3e2e367e/ios) to magnify parts of the screen. To allow them to resize the type, use [dynamic type](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/#dynamic-type-sizes), which allows them to resize the type in your app.

Reviewer: ```suggestion
In iOS, users can use [Full Screen Zoom or Window Zoom](https://support.apple.com/guide/iphone/zoom-iph3e2e367e/ios) to magnify parts of the screen. To allow them to resize the type, use [dynamic type](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/#dynamic-type-sizes), which enables your app's type to be resized.
```

Author: Thank you! I'll be sure to update authors.json

> If you manually limit how large your text scales it may cause issues for people who need a bigger font-- you want the text to scale dynamically instead of stopping at the point you delimit, because that size may still not be visible to all people.

Reviewer: ```suggestion
If you manually limit how large your text scales it may cause issues for people who need a bigger font. You want the text to scale dynamically instead of stopping at the point you set, because that size may still not be visible to all people.
```

Simplifying the language a tad here.

Author: Thank you!

[Minimum font-size](https://uxdesign.cc/guide-for-designing-better-mobile-apps-typography-5796495ef86f) for body text should be 17pt.

## Considerations for Android

Again, users can only zoom in on specified areas, so you'll need to use [scale-independent pixels](https://www.pixel-ruler.net/android-scale-independent-pixel) to determine the size of your text.

> There isn't a tool like Dynamic Type in Android, so you'll need to scale the text at different sizes yourself. Material design [has a resource](https://material.io/design/typography/the-type-system.html#type-scale) that makes a good starting point.

Reviewer: Maybe mention that it can be helpful to use an emulated version of your app to see how it better works in context?

Author: Committed a couple sentences to address this!
